volatility3/framework/objects/utility.py

```
@@ -102,6 +101,64 @@ def pointer_to_string(
     )
 
 
+def gather_contiguous_bytes_from_address(layer, address: int, count: int) -> bytes:
+    """
+    This method reconstructs a string from memory while also carefully examining each page
+
+    It goes page-by-page reading the bytes. This is done by calculating page boundaries
+    and then only reading one page at a time.
+
+    If a page is missing, the code initially catches the exception.
+    If data is non-empty (meaning at least one read succeeded), then we return what was read
+    If the first page fails, then we re-raise the exception
+    """
+
+    data = b""
+
+    left_to_read = count
+
+    # read as many pages as possible that are contiguous
+    # if the first page is missed, we re-raise the InvalidAddressException
+    # if we have at least 1 page that was read succesfully,
+    # then we try to construct a string from it
+    while left_to_read > 0:
+        # compute aligned address of current page and next the page
+        aligned = address & ~0xFFF
```

This is hard coding pages size, I think the layers have [functions](https://github.com/volatilityfoundation/volatility3/blob/d82200f9f747c38149a3f55f3453be0f88b51679/volatility3/framework/layers/intel.py#L95) to do this so that we don't need to hardcode them?  (Think it only works on intel pages).

Why not just get the mapping back (ignoring errors so it'll map with gaps) and then you can read as much as is needed and then barf if the next chunk isn't contiguous with the previous one?  Saves even needing to know about page values at all, and more efficient if you're reading from a large page for some reason?